Backpack/FloatingNotification/Classes/FloatingNotificationAnimator.swift

```diff
@@ -0,0 +1,74 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+
+protocol FloatingNotificationAnimatorDelegate: AnyObject {
+    func upAnimation()
+    func downAnimation()
+    func animationDidFinish()
+}
+
+final class FloatingNotificationAnimator: NSObject {
```

Comment:
does this need to be an NSObject?

Reply:
Yes, it is needed for the `perform` delay. FYI, the perform delay is used because if we use the animator delay then animation (not visible) starts too soon in the background and does not allow interaction such as the button tap.